eldritch/assets.py

```diff
@@ -273,6 +273,44 @@ def get_trigger(self, event, owner, state):
     return None
 
 
+class BlessingOrCurse(Card):
+  def __init__(self, kind, idx):
+    assert kind in ["Blessing", "Curse"]
+    super().__init__(kind, idx, "specials", {}, {})
+    self.opposite = "Curse" if kind == "Blessing" else "Blessing"
+    self.must_roll = False
+
+  def get_trigger(self, event, owner, state):
+    if isinstance(event, events.UpkeepActions) and event.character == owner:
+      if self.must_roll:
+        roll = events.DiceRoll(owner, 1)
+        return events.Conditional(
+            owner, roll, "sum",
+            {1: events.DiscardSpecific(owner, self), 2: events.Nothing()}
+        )
+      self.must_roll = True
```

Comment:
I have a bad feeling that get_trigger may be called more than once. Can you add some tests that verify that the don't-roll-the-first-turn part is working?
You could also implement this as a token, and just return an AddToken event if the current count is 0 (and the roll to discard otherwise).